File: CountOccurance_day_4/countFrequency.py
# Brute fore method 

# Time complexity : O(n^2)
# Space complexity : O(n)
arr = [1,2,3,4,2,4,1]

def countOccurrence(arr):
    seen = []
    result ={}
    count = 0

    for i in range(len(arr)):
        if arr[i] in seen:
            continue

        for j in range(len(arr)):
            if arr[i] == arr[j]:
                count += 1
        result[arr[i]] = count
        seen.append(arr[i])

    return arr[i]

result = countOccurrence(arr)
print(result)
    

File: CountOccurance_day_4/countFrequency.js
// Brute fore method 

// Time complexity : O(n^2)
// Space complexity : O(n) 

arr = [1,1,2,3,4,5,6,4]

function countFrequency(arr){
    let seen = [];
    let result = {}

    for(let i=0; i<arr.length; i++){
        if(seen.includes(arr[i])){
            continue;
        }
    let count = 0

    for(let j =0; i,arr.length; j++){
        if(arr[i] === arr[j]){
            count ++
        }
    }

    result[arr(i)] = count;
    seen.push(arr[i])
    }

    return result 

}


// Optimized approach


function countOcc(arr){

    const frequency = {}

    for(num of arr){
        if(frequency[num]){
            frequency[num]++
        }
        else frequency[num] = 1
    }

    return frequency

}


// v2

function countOccV2(arr){
    const frequency = {}

    for(num of arr){
        frequency[num] = (frequency[num] || 0) + 1 
    }
    return frequency
}


// Sorting

function countFreq(arr){
    arr.sort((a,b) => a-b) ;
        const frequency = {}

    for(num in arr){
        frequency[num] = (frequency[num] || 0) + 1
    }
    return frequency
}